cmd/chains.go

```diff
@@ -0,0 +1,123 @@
+package cmd
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	db "github.com/cosmos/cosmos-db"
+)
+
+type BlockStatePruner func(blockStoreDB, stateStoreDB db.DB, pruneHeight uint64) error
+
+// custom application.db pruner
+type AppPruner func(appStore db.DB, pruneHeight uint64) error
+
+// ChainPruner holds the specific pruning functions for a chain.
+type ChainPruner struct {
+	PruneBlockState BlockStatePruner
+	PruneApp        AppPruner
+}
+
+func pruneInjectiveAppStore(appStore db.DB, pruneHeight uint64) error {
+	logger.Info("Running Injective-specific application pruner")
+
+	if err := PruneAppState(appStore, pruneHeight); err != nil {
+		return err
+	}
+
+	if err := pruneAppStore(appStore, pruneHeight); err != nil {
+		return err
+	}
+
+	prefixesToPrune := map[string]int{
+		"s/k:oracle/s": 12,
```

Comment:
i don't trust keeping the lengths manually aligned -- can you make this an array of prefixes, and we calculate length instead? it's not like it saves a lot of compute